core/src/main/scala/org/apache/spark/sql/execution/CompactExecRowToMutableRow.scala

```diff
@@ -31,24 +31,16 @@ import org.apache.spark.sql.types.{DataType, Decimal, DecimalType, StructType}
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.unsafe.types.UTF8String
 
-trait CompactExecRowToMutableRow extends ResultWasNull {
+abstract class CompactExecRowToMutableRow extends ResultNullHolder {
```

Comment:
The wasNull interface is part of ResultSet which is set only after a read of a column to tell whether previous read returned a null value or not. Required for primitive values at least that cannot be tested for nullability otherwise.

We carried over similar into AbstractCompactExecRow read methods which accepts a "ResultWasNull" object to set into it whether the previous read returned a null value. Normally EmbedResultSet implements that interface so that it can return appropriate value for subsequent wasNull() call. However, here we are reading the AbstractCompactExecRow directly hence creating a holder object that can be passed to its getAs* methods (either CompactExecRowToMutableRow or created directly in RowTableScan generated code when using raw region iterator).